src/all.rs

```diff
@@ -57,6 +125,58 @@ impl<'a, Trait: ?Sized + TraitQuery> Iterator for ReadTableTraitsIter<'a, Trait>
     }
 }
 
+impl<'a, Trait: ?Sized + TraitQuery> Iterator for AddedReadTableTraitsIter<'a, Trait> {
+    type Item = &'a Trait;
+    fn next(&mut self) -> Option<Self::Item> {
+        // Iterate the remaining table components that are registered,
+        // until we find one that exists in the table.
+        let (column, meta) = unsafe { zip_exact(&mut self.components, &mut self.meta) }
+            .find_map(|(&component, meta)| self.table.get_column(component).zip(Some(meta)))?;
+
+        // SAFETY: we know that the table row is a valid index???
+        let column_ticks = unsafe { column.get_ticks_unchecked(self.table_row).deref() };
+        column_ticks
+            .is_added(self.last_change_tick, self.change_tick)
+            .then(|| unsafe {
+                // SAFETY ISSUE! SAFETY ISSUE! SAFETY ISSUE! Unlike in the write case, we do not have
+                // exclusive access! We have shared access to the entire column and ticks? This might be
+                // okay though because there cannot be any other accesses with write access at the same
+                // time?
+                let ptr = column
+                    .get_data_ptr()
+                    .byte_add(self.table_row * meta.size_bytes);
+                meta.dyn_ctor.cast(ptr)
+            })
+            .or_else(|| self.next())
```

Comment:
We should use loops instead of recursion here, though this PR is already really big so we can leave this change for a followup if you prefer.